### PR TITLE
NDS-1203 Refactor modal to not map buttons

### PR DESCRIPTION
**Pull request by matthewlyle**
Modals no longer accept `primaryButton` or `secondaryButtons` as props and instead have `footerContent` that can accept a React node, allowing more flexibility in the content of a Modal's footer.

  Old API:

  ```
<Modal 
	primaryButton={{ label: "Primary Action", onClick: () => {} }} 
	secondaryButtons={[{ label: "Secondary Action", onClick: () => {} }]} 
/>
  ```

  New API:

  ```
const modalButtons = (
	<ButtonGroup>
		<PrimaryButton onClick={}>Primary Action</PrimaryButton>
		<QuietButton onClick={}>Secondary Action</QuietButton>
	</ButtonGroup>
);

<Modal footerContent={modalButtons}>
  ```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,14 +5,42 @@ All notable changes to this project will be documented in this file.
...
+
 ### Changed
+
+- **BREAKING CHANGE** [Modals](nulogy.design/components/Modal/) no longer accept `primaryButton` or `secondaryButtons` props and instead have `footerContent` that can accept a React node, allowing more flexibility in the content of a Modal's footer.
+
+  Old API:
+
+  ```
+  <Modal
+
```
Comment (marboro92) [inline CHANGELOG.md]: missing primaryButton, and secondaryButtons props in the code sample here?

File: components/src/Modal/Modal.story.js [hunk 2]
```diff
@@ -173,13 +136,7 @@ storiesOf("Modal", module)
...
   ))
   .add("with select and scrolling content", () => (
-    <Modal
-      title="Edit Profile"
-      onRequestClose={() => {}}
-      primaryButton={{ label: "Submit", type: "submit", form: "myForm" }}
-      secondaryButtons={[{ label: "Cancel", onClick: () => {} }]}
-      maxWidth="456px"
-    >
+    <Modal title="Edit Profile" footerContent={modalButtons} onRequestClose={() => {}} maxWidth="456px">
```
Comment (marboro92) [inline components/src/Modal/Modal.story.js:144]: Should there be a story with no footerContent just so we capture that use case visually?

**Review by marboro92 — APPROVED** [review]

File: components/src/Modal/Modal.story.js (hunk 2 above)
Comment (matthewlyle) [reply components/src/Modal/Modal.story.js:144]: Yes.

File: CHANGELOG.md (hunk 1 above)
Comment (matthewlyle) [reply CHANGELOG.md]: lol wtf where did it go... will add back now...

**Pull request merged**